### DISCUSSION TITLE
Difference between Laravel validation.php and Laravel Lang validation.php

### DISCUSSION THREAD

**Discussion by fabius85** [Laravel-Lang/common#28] [Q&A]
### Description:
Looks like [source/validation.php](https://github.com/Laravel-Lang/lang/blob/master/source/validation.php) is outdated if compared to current 8.x Laravel [resources/lang/en/validation.php](https://github.com/laravel/laravel/blob/8.x/resources/lang/en/validation.php)
This difference is then reflected also on [source/validation-inline.php](https://github.com/Laravel-Lang/lang/blob/master/source/validation-inline.php)

The following strings are different:
`alpha`, `alpha_dash`, `alpha_num`, `gte` [`numeric`, `file`, `string`], `lte` [`numeric`, `file`, `string`], `max` [`numeric`, `file`, `string`, `array`], `multiple_of`

The following strings are not present in the original Laravel `validation.php`:
`attached`, `relatable`

The following entire block at the end of the file is missing from Laravel Lang `validation.php`:
```
    /*
    |--------------------------------------------------------------------------
    | Custom Validation Attributes
    |--------------------------------------------------------------------------
    |
    | The following language lines are used to swap our attribute placeholder
    | with something more reader friendly such as "E-Mail Address" instead
    | of "email". This simply helps us make our message more expressive.
    |
    */

    'attributes' => [],
```

Is there any reason for not updating it to match original Laravel `validation.php`?

**Comment by caouecs** [comment]
Laravel 6/7

**Reply by fabius85** [reply]
Ok so removing the `attached` and `relatable` keys would break compatibility with previous Laravel versions.
But I think that updating the english validation strings with new ones from Laravel's `validation.php` shouldnt do any harm.
I mean, the array keys are the same, it's just the message that was slightly modified in a more proper english.
Example for `gte.numeric`:
old message > 'The :attribute must be greater than or equal :value.'
new message > 'The :attribute must be greater than or equal **to** :value.'

**Reply by andrey-helldar** [reply]
> Ok so removing the `attached` and `relatable` keys would break compatibility with previous Laravel versions.

The `attached` and `relatable` keys are not used in Laravel 6, 7, 8 and 9:

- https://github.com/laravel/laravel/blob/6.x/resources/lang/en/validation.php
- https://github.com/laravel/laravel/blob/7.x/resources/lang/en/validation.php
- https://github.com/laravel/laravel/blob/8.x/resources/lang/en/validation.php
- https://github.com/laravel/laravel/blob/master/lang/en/validation.php

[PR](https://github.com/Laravel-Lang/lang/pull/1935) to remove these keys has already been created.

**Reply by andrey-helldar** [reply]
As for correcting the proposal forms to be more correct, you can create a PR.

**Reply by andrey-helldar** [reply]
The `attached` and `relatable` keys belong to Nova.

![image](https://user-images.githubusercontent.com/10347617/149297439-f26b91ee-88b2-45a8-81cd-6dfd94fbd3d7.png)

**Reply by andrey-helldar** [reply]
@caouecs, I created a https://github.com/Laravel-Lang/lang/pull/1940 on splitting the main validation files from Nova.

**Comment by andrey-helldar** [comment]
@fabius85

> The following entire block at the end of the file is missing from Laravel Lang `validation.php`:

That's right. Since we use two versions of translations, regular and inline, we decided to move the block with [`attribute`](https://github.com/Laravel-Lang/lang/blob/master/source/validation-attributes.php) names to an external file.

This will avoid duplication of data when using [`translation managers`](https://github.com/Laravel-Lang/publisher), since translation of attributes may be needed when using any of the files.
